source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -270,6 +288,32 @@ void AuthenticatorImpl::verifyKey() {
   handleGoodJwt(/*cache_hit=*/false);
 }
 
+void AuthenticatorImpl::addJWTClaimToHeader(const std::string& claim_name,
+                                            const std::string& header_name) {
+  StructUtils payload_getter(jwt_->payload_pb_);
+  const ::Envoy::ProtobufWkt::Value* claim_value;
+  const auto status = payload_getter.GetValue(claim_name, claim_value);
+  std::string str_claim_value;
+  if (status == StructUtils::OK) {
+    if (claim_value->kind_case() == Envoy::ProtobufWkt::Value::kStringValue) {
+      str_claim_value = claim_value->string_value();
+    } else if (claim_value->kind_case() == Envoy::ProtobufWkt::Value::kNumberValue) {
+      str_claim_value = convertClaimDoubleToString(claim_value->number_value());
+    } else if (claim_value->kind_case() == Envoy::ProtobufWkt::Value::kBoolValue) {
+      str_claim_value = claim_value->bool_value() ? "true" : "false";
+    } else {
+      ENVOY_LOG(debug,
+                "--------claim : {} is not a primitive type of int, string, or bool -----------",
+                claim_name);
```

Comment:
There are many comments so that comment for lost somewhere. @qiwzhang responded to your comment with 
`I feel that adding stat for these cases will not be so useful. The symptoms of such errors will show up in other ways. In this case, the backend will expect certain claim headers, but due some mistakes, such as claim name typo or wrong type, we will not copy them, the backend will not receive them, it will complain. They can be easily found in the development test as JWT payload format usually don't change.`